python/pyarrow/types.pxi

```diff
@@ -5236,6 +5273,43 @@ def run_end_encoded(run_end_type, value_type):
     return pyarrow_wrap_data_type(ree_type)
 
 
+def json_(DataType storage_type=utf8()):
+    """
+    Create instance of JSON extension type.
+
+    Parameters
+    ----------
+    storage_type : DataType
+        The underlying data type.
```

Comment:
Maybe mention here that this should be one of the string data types? (also, do you get a proper error message when passing a non-string data type? Can you test that?)